packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/rest_api/FlutterRestApi.kt

```diff
@@ -0,0 +1,269 @@
+            var apiName: String?
+            var options: RestOptions
+
+            try {
+                cancelToken = FlutterApiRequest.getCancelToken(request)
+                apiName = FlutterApiRequest.getApiPath(request)
+                options = FlutterApiRequest.getRestOptions(request)
+            } catch (e: Exception) {
+                FlutterApiError.postFlutterError(
+                        flutterResult,
+                        FlutterApiErrorMessage.AMPLIFY_REQUEST_MALFORMED.toString(),
+                        e)
+                return
+            }
+
+            try {
+                var operation: RestOperation?
+                if (apiName == null) {
+                    operation = functionWithoutApiName(options,
+                            Consumer { result ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareRestResponseResult(flutterResult, result, methodName) },
+                            Consumer { error ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareError(
+                                        flutterResult,
+                                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                                        cancelToken,
+                                        error)
+                            }
+                    )
+                } else {
+                    operation = functionWithApiName(
+                            apiName,
+                            options,
+                            Consumer { result ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareRestResponseResult(flutterResult, result, methodName) },
+                            Consumer { error ->
+                                if (!cancelToken.isNullOrEmpty()) OperationsManager.removeOperation(cancelToken)
+                                prepareError(
+                                        flutterResult,
+                                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                                        cancelToken,
+                                        error)
+                            }
+                    )
+                }
+                OperationsManager.addOperation(cancelToken, operation!!)
+
+            } catch (e: Exception) {
+                prepareError(
+                        flutterResult,
+                        FlutterApiErrorMessage.stringToAPIRestError(methodName).toString(),
+                        cancelToken,
+                        e)
+            }
+        }
+
+        fun prepareError(flutterResult: Result, msg: String, cancelToken: String, error: Exception) {
```

Comment:
Why are we passing the cancel token here?
Also can we not call `FlutterApiError.postFlutterError()` where `prepareError()` is called? Is there a need for this redirection.